src/sentry/static/sentry/app/components/asyncComponent.jsx

```diff
@@ -79,14 +93,15 @@ class AsyncComponent extends React.Component {
     this.setState(this.getDefaultState(), this.fetchData);
   };
 
-  fetchData = () => {
+  visibilityReloader = () => !document.hidden && this.reloadData();
+
+  reloadData = () => this.fetchData({reloading: true});
```

Comment:
We need a way for subclassed Component to control when to reload, e.g. we don't want to reload every time the user alt-tabs back and forth. I think we probably want another instance variable and by default this doesn't reload data when `reloadOnVisible` is true.

Reply:
Why not just mutate the `reloadOnVisible` instance variable in that case?

Reply:
Technically that would work, but it conflates the meaning of `reloadOnVisible` - this feature only works if its value is `true` at the time of mounting, and it's confusing to also mutate throughout its lifecycle.

Reply:
Most importantly, by default, it should require implementor to toggle a flag in order to have the reload work (I think a 2nd instance variable makes more sense)

Reply:
> this feature only works if its value is true at the time of mounting

Ah yeah totally true, I overlooked this. I'll add a 2nd toggle flag